prboom2/src/dsda/dgeom.h

```diff
@@ -0,0 +1,212 @@
+//
+// Copyright(C) 2023 Brian Koropoff
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	Internal double-precision geometry functions
+//
+//      Not intended for use outside of DSDA subdir, so these aren't
+//      prefixed with dsda_
+//
+
+#ifndef DSDA_INTERNAL_
+#error Internal header, not for use outside dsda subdir
+#endif
+
+#ifndef DSDA_DGEOM_H_
+#define DSDA_DGEOM_H_
+
+#include <math.h>
+
+#include "m_fixed.h"
+#include "r_defs.h"
+#include "r_state.h"
+
+struct dpoint
+{
+  double x, y;
+};
+
+// Can be a line (infinitely long) given by the points
+// or the segment between start and end
+struct dline
+{
+  struct dpoint start, end;
+};
+
+// Give names to sides to make code very explicit
+enum side
+{
+  SIDE_RIGHT = 0,
+  SIDE_LEFT = 1
+};
+
+#ifndef M_SQRT2
+#define M_SQRT2 1.41421356237309504880
+#endif
+
+// Epsilon values used to "round" in various tests.  We don't rescale
+// fixed_t values when converting to double, so these are all relative
+// to FRACUNIT.
+// Most test functions take an explicit epsilon parameter which shadows
+// `epsilon` or `epsilonr2` to make it clear what default to use.
+#define EPSILON_FACTOR (1 << 14);
+// Ordinary epsilon, e.g. uncertainty in x or y
+static const double epsilon = (double) FRACUNIT / EPSILON_FACTOR;
+// Uncertainty in an x,y distance
+static const double epsilonr2 = (double) FRACUNIT * M_SQRT2 / EPSILON_FACTOR;
+// Uncertainty in a map x or y
+static const double epsilonmap = (double) FRACUNIT - 1;
+// Uncertainty in a map x,y distance
+static const double epsilonmapr2 = (double) FRACUNIT * M_SQRT2 - 1;
+
+//
+// Inline functions
+// 
+
+// Conversion functions
+
+static inline fixed_t DoubleToFixed(double v)
```

Comment:
What does this function do exactly? If it's to fixed, it should multiply by FRACUNIT. If the double is already in fixed units, another name would be preferable.

Reply:
Yeah, all the dgeom functions use fixed units to avoid dividing/multiplying on conversion

Reply:
I added a `dfixed_t` alias for double to make it clear that these functions are intended to work on fixed units.

Reply:
👍